### ISSUE TITLE
Integration of an online image editor

### ISSUE THREAD

**Issue by Man-in-Black** [BookStackApp/BookStack#4476]
### Describe the feature you'd like

It would be a nice feature if it were possible to edit an image right in Bookstack.
Therefore one will be able to make some markings within an image (which was taken with a camera for example).
If looked around and there is a promising project here: https://github.com/viliusle/miniPaint
Maybe it is possible to integrate this in a manner like the drawings.io integration.

### Describe the benefits this would bring to existing BookStack users

Simple edititing of pictures right out of bookstack.

### Can the goal of this request already be achieved via other means?

Can't find anything about it.

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundemental request

### How long have you been using BookStack?

0 to 6 months

**Comment by virtadpt** [comment]
Yes, please.  I found this request when searching through the tickets before opening my own.

Use case: For some of my research, I often take photographs with my phone of something I'm working on for the purpose of putting the images into pages in my Bookstack wiki.  The workflow to do this looks a little like this:

- Take pictures on my phone
- E-mail them to myself from my phone
- Save files locally
- Edit in the GIMP (crop and redact, usually)
- Save locally
- Insert into Bookstack page

It would be really handy to be able to do this without having to involve the GIMP or anything else prior to inserting the images (or even without having to get the images off of my phone - page, crop, and save would be great).  Being able to scale images from the browser is nice and all, but a little more functionality would be great.

**Comment by virtadpt** [comment]
I did some looking around, and there are some options that might be considered:

- https://github.com/scaleflex/filerobot-image-editor (I don't know how heavy this option is, all things considered)
- https://github.com/jimp-dev/jimp
- https://github.com/fengyuanchen/cropperjs (if nothing else, it makes cropping images possible instead of just resizing them)
- https://github.com/nhn/tui.image-editor
- https://github.com/MattKetmo/darkroomjs (Of course, it's been DV'd.)

**Comment by virtadpt** [comment]
If nothing else, this might help with #4875.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/4476
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
